### ISSUE TITLE
Add FocusOutAsync method to ElementReference in Blazor .NET 5 server side

### ISSUE THREAD

**Issue by Marius-Kazlauskas** [dotnet/aspnetcore#28127]
It already is FocusAsync method in ElementReference (Blazor .NET 5 server side), but is very missing FocusOutAsync, because for example I need to focus out input element when I press enter on it

**Comment by javiercn** [comment]
@Marius-Kazlauskas thanks for contacting us.

The browser doesn't offer a native "focusOut" method. It is not our goal to extend the set of primitives available on ElementReference beyond what HTML provides.

**Comment by Marius-Kazlauskas** [comment]
sorry, I had in mind blur() method, so it would be wonderful to have it here in the future

**Comment by SteveSandersonMS** [comment]
Since it's very easy to implement this through JS interop, it's likely we won't need to prioritise this.

**Comment by mckaragoz** [comment]
I agree with this issue. We already have FocusAsync and i think having also the opposite makes sense. And lots of UI libraries can use this without working on JS interop.

**Comment by taublast** [comment]
> Since it's very easy to implement this through JS interop, it's likely we won't need to prioritise this.

Why did MS implement FocusAsync then. It's like implementing method TurnToLeft, then when one asks why TurnToRight is not implemented he gets an answered like implement it yourself with JS.

**Comment by javiercn** [comment]
@taublast `HTMLElement.focus()` is a feature defined by the DOM APIs so we decided to expose it, however there is no similar functionality for `focusOut()` defined by the DOM. If we were to define such functionality in Blazor we would have two potential isues:
* We would have to define the semantics (who gets the focus).
* We would risk baking in a set of behaviors that might conflict with a potential future provided DOM API.

I do not think the TurnLeft/Right analogy is accurate here because there are many options to choose from when focusing out, and they are very likely app dependent. Which is likely why JavaScript only offers a `focus()` method and not a `focusOut()` method.

**Comment by mckaragoz** [comment]
@javiercn yes javascript not offer `focusout()` but it has `blur()`. I also opened an issue #34156.

I also prepared a PR to add BlurAsync into MudBlazor, and it was merged.
https://github.com/MudBlazor/MudBlazor/pull/4929
